TTS/utils/arguments.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+"""Argument parser for training scripts."""
+
+import argparse
+import re
+import glob
+import os
+
+from TTS.utils.generic_utils import (
+    create_experiment_folder, get_git_branch)
+from TTS.utils.console_logger import ConsoleLogger
+from TTS.utils.io import copy_model_files, load_config
+from TTS.utils.tensorboard_logger import TensorboardLogger
+
+from TTS.tts.utils.generic_utils import check_config_tts
+
+
+def parse_arguments(argv):
+    """Parse command line arguments of training scripts.
+
+    Parameters
+    ----------
+    argv : list
+        This is a list of input arguments as given by sys.argv
+
+    Returns
+    -------
+    argparse.Namespace
+        Parsed arguments.
+
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--continue_path",
+        type=str,
+        help=("Training output folder to continue training. Used to continue "
+              "a training. If it is used, 'config_path' is ignored."),
+        default="",
+        required="--config_path" not in argv)
+    parser.add_argument(
+        "--restore_path",
+        type=str,
+        help="Model file to be restored. Use to finetune a model.",
+        default="")
+    parser.add_argument(
+        "--config_path",
+        type=str,
+        help="Path to config file for training.",
+        required="--continue_path" not in argv)
+    parser.add_argument(
+        "--debug",
+        type=bool,
+        default=False,
+        help="Do not verify commit integrity to run training.")
+    parser.add_argument(
+        "--rank",
+        type=int,
+        default=0,
+        help="DISTRIBUTED: process rank for distributed training.")
+    parser.add_argument(
+        "--group_id",
+        type=str,
+        default="",
+        help="DISTRIBUTED: process group id.")
+
+    return parser.parse_args()
+
+
+def get_last_checkpoint(path):
+    """Get latest checkpoint from a list of filenames.
+
+    It is based on globbing for `*.pth.tar` and the RegEx
+    `checkpoint_([0-9]+)`.
+
+    Parameters
+    ----------
+    path : list
+        Path to files to be compared.
+
+    Raises
+    ------
+    ValueError
+        If no checkpoint files are found.
+
+    Returns
+    -------
+    last_checkpoint : str
+        Last checkpoint filename.
+
+    """
+    last_checkpoint_num = 0
+    last_checkpoint = None
+    filenames = glob.glob(
+        os.path.join(path, "/*.pth.tar"))
+    for filename in filenames:
+        try:
+            checkpoint_num = int(
+                re.search(r"checkpoint_([0-9]+)", filename).groups()[0])
+            if checkpoint_num > last_checkpoint_num:
+                last_checkpoint_num = checkpoint_num
+                last_checkpoint = filename
+        except AttributeError:  # if there's no match in the filename
+            pass
+    if last_checkpoint is None:
+        raise ValueError(f"No checkpoints in {path}!")
+    else:
+        return last_checkpoint
```

Comment:
I've added this for step based model continuing. Currently it doesn't work so I've commented it out in `process_args()` as the best model is not taken into account. The currently used creation time is a bit volatile, so I've started a discussion on this topic in  #643 :wink:

Reply:
how about if we separate it from this PR? We can discuss that as I merge the rest.

Reply:
Yes yes absolutely :+1: